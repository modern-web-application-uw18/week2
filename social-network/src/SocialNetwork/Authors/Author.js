import React, { Component } from 'react';


class Author extends Component {

    render(){
        const authorInfo = this.props.authorInfo;
        console.log(authorInfo);
        return(
        
            
            
              //authorInfo.map(function(author){
                //return (
                <div className = "authorSection">
                    <div><img src={authorInfo.image} className = "authorImage" /></div>
                    <div>
                        <div><a href = {authorInfo.link}>{authorInfo.name}</a></div>
                        <div>{authorInfo.minutesToRead} min read</div>
                    </div>
                    
                </div>
                //)
              //})
            
            
        );    
        
    }
}

export default Author;