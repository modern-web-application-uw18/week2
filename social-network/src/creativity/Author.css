.Author-image-container {
    height: 50px;
    width: 50px;
    padding-top: 5px;
    margin-left: 5px;
}

.Author-image-member-container {
    height: 50px;
    width: 50px;
    padding-top: 5px;
    margin-left: 5px;
    background-image: url('../img/arcs.png');
}

.Author-image-container img {
    width: 40px;
    border-radius: 50%;
    overflow: hidden;
}

.Author-image-member-container img {
    width: 40px;
    border-radius: 50%;
    overflow: hidden;
}

.Author-text-container {
    display: flex;
    flex-wrap: nowrap;
    flex-direction: column;
    color: gray;
    height: 56px;
    width: 150px;
    padding-left: 10px;
    padding-top: 10px;
}

.Author-name {
    font-size: 0.75em;
    color: black;
    text-align: left;
    font-weight: 600;
    height: 20px;
}

.Author-post {
    font-size: 0.65em;
    text-align: left;
    font-weight: 500;
    height: 20px;
}

.Article-bookmark-container {
    height: 25px;
    width: 25px;  
    padding-top: 10px;
    margin-left: 10px;
}

.Article-bookmark-container img {
    height: auto;
    width: 90%;  
}
