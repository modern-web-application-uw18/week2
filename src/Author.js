import React from 'react';

const Author = ({ author: {name, image, isMediumMember} }) =>

(
  <div>
    <img className='authorImage' src={image} alt='' />
    <div>
      <h6>{name}</h6>

    </div>
  </div>
)

export default Author;

// <h6>{postedDate}</h6>
