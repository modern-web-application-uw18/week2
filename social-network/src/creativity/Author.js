import React, { Component } from 'react';
import PropTypes from 'prop-types';
import './Author.css';
import bookmark from '../img/icon-bookmark.svg';

class Author extends Component {

    render() {

        const {
            name,
            image,
            isMediumMember
        } = this.props.author;

        const showMember = (member) => {
            if (member) {
                return (
                    <div className="Author-image-member-container">
                        <img src={image} alt="" />
                    </div>
                )
            } else {
                return (
                    <div className="Author-image-container">
                        <img src={image} alt="" />
                    </div>
                )
            }
        }
        
        const formatDate = (date) => {
            const monthNames = ['', 'Jan', 'Feb', 'Mar', 'Apr', 'May', 'Jun', 'Jul', 'Aug', 'Sep', 'Oct', 'Nov', 'Dec'];
            let dateTime = date.split('T');
            let dateFields = dateTime[0].split('-');
            return monthNames[parseInt(dateFields[1], 10)] + ' ' + dateFields[2];
        }

        return (
            <div className="Author">
		        {showMember(isMediumMember)}
                <div className="Author-text-container">
                    <div className="Author-name">{name}</div>
                    <div className="Author-post">{formatDate(this.props.postdate)} &middot; {this.props.minutes} min read</div>
                </div>
                <div className="Article-bookmark-container">
                    <img src={bookmark} alt="Bookmark" />
                </div>
            </div>
        );
    }
}

Author.propTypes = {
    name: PropTypes.string.isRequired,
    image: PropTypes.string,
    ismember: PropTypes.bool.isRequired,
    postdate: PropTypes.string.isRequired,
    minutes: PropTypes.number.isRequired
};

Author.defaultProps = {
    name: 'Anonymous',
    ismember: false
};

export default Author;