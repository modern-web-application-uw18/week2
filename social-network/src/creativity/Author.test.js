import React from 'react';
import ReactDOM from 'react-dom';
import TestRenderer from 'react-test-renderer';
import Author from './Author';

describe('AuthorInfo', () => {
    it('renders without crashing', () => {
        const div = document.createElement('div');
        const author = {name: 'Jerry Mathers',
                        image: 'https://cdn-images-1.medium.com/fit/c/80/80/1*wsP97tix16onOIkq4STxuw.jpeg',
                        isMediumMember: false};
        ReactDOM.render(<Author
            author={author}            
            postdate='2018-05-24T07:00:00.000Z'
            minutes={3} />, div);
        ReactDOM.unmountComponentAtNode(div);
    });

    it('should render with author name, article post date, etc.', () => {
        const author = {name: 'Jerry Mathers',
                        image: 'https://cdn-images-1.medium.com/fit/c/80/80/1*wsP97tix16onOIkq4STxuw.jpeg',
                        isMediumMember: false};
        const component = TestRenderer.create(<Author
                author={author}            
                postdate="2017-06-11T07:00:00.000Z"
                minutes={3} />);
        expect(component.toJSON()).toMatchSnapshot();
    });
});