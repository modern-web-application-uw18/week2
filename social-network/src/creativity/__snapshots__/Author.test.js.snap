// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AuthorInfo should render with author name, article post date, etc. 1`] = `
<div
  className="Author"
>
  <div
    className="Author-image-container"
  >
    <img
      alt=""
      src="https://cdn-images-1.medium.com/fit/c/80/80/1*wsP97tix16onOIkq4STxuw.jpeg"
    />
  </div>
  <div
    className="Author-text-container"
  >
    <div
      className="Author-name"
    >
      Jerry Mathers
    </div>
    <div
      className="Author-post"
    >
      Jun 11
       Â· 
      3
       min read
    </div>
  </div>
  <div
    className="Article-bookmark-container"
  >
    <img
      alt="Bookmark"
      src="icon-bookmark.svg"
    />
  </div>
</div>
`;
